### ISSUE TITLE
libApp.getMeta()

### ISSUE THREAD

**Issue by ComLock** [enonic/xp#7285]
I think that on the java layer applications can call java services between processes (IPC).
As far as I know there is no such thing on the javascript layer, and Enonic isn't going to implement that (for good reasons).

I would like the possibility to fetch meta data from an application.

@alansemenov told me we already have one such meta data: "system application" which is used to filter the list of applications under Applications administrator.

Specifically I would like to mark whether an application has collectors or not.
And perhaps filter libApp.list() on that property (see enonic/xp#6671).

In addition I would like to provide a list of collectors, location of their react bundles and collector task. 

So basically I would like to provide a meta info json at compile time which can be read from other apps runtime.

**Comment by sigdestad** [comment]
So, you basically need an extended API for more info about the site config?

**Comment by sigdestad** [comment]
Or are you looking for some kind of dependency system?

**Comment by ComLock** [comment]
Not sites. This has nothing to do with CMS. It's about apps. App name is not enough info. I need some way of tagging apps with information and lookup that information.

**Comment by alansemenov** [comment]
@rymsha Can you check what kind of info we are getting from manifest in Java API? Can developer add custom fields to the manifest and then get their values from Java API?

**Comment by rymsha** [comment]
lib-io has a method `getResource`, it is capable to get file (json file, whatever) from a known location of any currently running app! Magic part is that it accepts either `string` (only goes for resources in current app) or `ResourceKey` than it can grab any application's resource. Sure, not well documented (probably because it is a niche feature) but exists since Y2015

**Comment by alansemenov** [comment]
@ComLock is it good enough to solve the issue?

**Comment by ComLock** [comment]
getResource can only be read from within the same app, so nope.

**Comment by rymsha** [comment]
> getResource can only be read from within the same app, so nope.

Actually
> lib-io has a method getResource, it is capable to get file (json file, whatever) from a known location of any currently running app!

**Comment by alansemenov** [comment]
@rymsha what do you mean by "a known location of any currently running app"? location within the jar? absolute path? do you have an example?

**Comment by rymsha** [comment]
```
    const ioLib = require('/lib/xp/io');
    const appKey = 'my.cool.app'; // some "currently running app"
    const filePath = '/some/file.json'; // "known location" inside `my.cool.app`
    const resource = ioLib.getResource(Java.type('com.enonic.xp.resource.ResourceKey').from(appKey + ':' + filePath));
    log.info(ioLib.readText(resource.getStream()));
```

**Comment by ComLock** [comment]
Eh, so these issues I report. Is the goal always to show the first workaround we find, just for my usecase, and not make a solution accessible to normal js programmers?

**Comment by alansemenov** [comment]
Depends on a use case. This is a very specific one which will not only require creation of a new lib/API we don't have (lib-app) but also implementation of a potentially unsafe communication between applications in this public API.

We can keep this issue in a backlog for a future discussion but if you need this now then I suggest you try a workaround suggested by @rymsha

**Comment by ComLock** [comment]
Trying it out now.

**Comment by ComLock** [comment]
Yup the code works to get a resource from another app. However since I don't know what the app names are I can't use this workaround yet. I need some way to get a list of running apps: https://github.com/enonic/xp/issues/6671

**Comment by ComLock** [comment]
Works https://github.com/enonic/lib-explorer/blob/master/src/main/resources/lib/explorer/collector/list.es#L20
